nginx-controller/controller/controller.go

```diff
@@ -1067,3 +1180,110 @@ func (lbc *LoadBalancerController) ValidateSecret(secret *api_v1.Secret) error {
 
 	return fmt.Errorf("Secret is not a TLS or JWK secret")
 }
+
+// getMinionsForHost returns a list of all minion ingress resources for a given master
+func (lbc *LoadBalancerController) getMinionsForMaster(master nginx.IngressEx) ([]nginx.IngressEx, error) {
+	ings, err := lbc.ingLister.List()
+	if err != nil {
+		return []nginx.IngressEx{}, err
+	}
+
+	minions := []nginx.IngressEx{}
+	for i, _ := range ings.Items {
+		if !lbc.isNginxIngress(&ings.Items[i]) {
+			continue
+		}
+		if ings.Items[i].Annotations["nginx.org/mergible-ingress-type"] == "minion" {
+			if len(ings.Items[i].Spec.Rules) != 1 {
+				glog.Errorf("Ingress Resource %s with the 'nginx.org/mergible-ingress-type' annotation must contain only one host", ings.Items[i].Name)
+				continue
+			}
+			if ings.Items[i].Spec.Rules[0].HTTP == nil {
+				glog.Errorf("Ingress Resource %s with the 'nginx.org/mergible-ingress-type' annotation set to 'minion' must contain a Path", ings.Items[i].Name)
+				continue
+			}
+
+			ingEx, err := lbc.createIngress(&ings.Items[i])
+			if err != nil {
+				glog.Errorf("Error creating ingress resource %v: %v", ingEx.Ingress.Name, err)
+				continue
+			}
+			if ingEx.Ingress.Spec.Rules[0].Host == master.Ingress.Spec.Rules[0].Host {
+				if len(ingEx.TLSSecrets) > 0 || ingEx.JWTKey != nil {
+					glog.Errorf("Ingress Resource %s with the 'nginx.org/mergible-ingress-type' annotation set to 'minion' cannot contain TLSSecrets or JWTKeys", ingEx.Ingress.Name)
+					continue
+				}
+				minions = append(minions, *ingEx)
+			}
+		}
+	}
+	return minions, nil
+}
+
+// findMasterForHost returns a master for a given minion
+func (lbc *LoadBalancerController) findMasterForMinion(minion extensions.Ingress) (extensions.Ingress, error) {
+	ings, err := lbc.ingLister.List()
+	if err != nil {
+		return extensions.Ingress{}, err
+	}
+
+	for i, _ := range ings.Items {
+		if !lbc.isNginxIngress(&ings.Items[i]) {
+			continue
+		}
+		if !lbc.cnf.HasIngress(&ings.Items[i]) {
+			continue
+		}
+		ingEx, err := lbc.createIngress(&ings.Items[i])
```

Comment:
in this method we don't need to call createIngress -- we only need to find the Ingress, not to validate it

Reply:
Done.